File: easy/intersect.go
package main

import (
    "os"
    "bufio"
    "fmt"
    "strings"
)

func contains(a []string, b string) bool {
    for _, c := range a {
        if c == b {
            return true
        }
    }
    return false
}

func main() {
    f, _ := os.Open(os.Args[1])
    r := bufio.NewReader(f)
    for {
        l, err := r.ReadString('\n')
        if err != nil { break }
        l = strings.TrimSpace(l)
        arr := strings.Split(l, ";")
        x := strings.Split(arr[0], ",")
        y := strings.Split(arr[1], ",")
        var match []string
        for _, a := range x {
            if contains(y, a) {
                match = append(match, a)
            }
        }
        fmt.Println(strings.Join(match, ","))
    }
}

File: easy/intersect.py
import sys

if __name__ == "__main__":
    f = open(sys.argv[1])
    for l in f:
        x, y = l.rstrip().split(';')
        x = set(int(t) for t in x.split(','))
        y = set(int(t) for t in y.split(','))
        print ','.join(str(t) for t in x.intersection(y))
